src/main/java/net/minecraftforge/common/loot/LootModifierManager.java

```diff
@@ -0,0 +1,80 @@
+package net.minecraftforge.common.loot;
+
+import java.util.Collection;
+import java.util.HashMap;
+//import static net.minecraft.client.resources.JsonReloadListener.LOGGER;
+
+import java.util.Map;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableMap.Builder;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonObject;
+
+import net.minecraft.client.resources.JsonReloadListener;
+import net.minecraft.profiler.IProfiler;
+import net.minecraft.resources.IResourceManager;
+import net.minecraft.server.MinecraftServer;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.world.storage.loot.conditions.ILootCondition;
+import net.minecraft.world.storage.loot.conditions.LootConditionManager;
+import net.minecraft.world.storage.loot.functions.ILootFunction;
+import net.minecraft.world.storage.loot.functions.LootFunctionManager;
+import net.minecraftforge.fml.LogicalSide;
+import net.minecraftforge.fml.common.thread.EffectiveSide;
+import net.minecraftforge.registries.ForgeRegistries;
+
+public class LootModifierManager extends JsonReloadListener {
+    private static LootModifierManager instance;
+    public static final Logger LOGGER = LogManager.getLogger();
+    private static final Gson GSON_INSTANCE = (new GsonBuilder()).registerTypeHierarchyAdapter(ILootFunction.class, new LootFunctionManager.Serializer()).registerTypeHierarchyAdapter(ILootCondition.class, new LootConditionManager.Serializer()).create();
+
+    private Map<ResourceLocation, IGlobalLootModifier> registeredLootModifiers = ImmutableMap.of();
+
+    public LootModifierManager() {
+        super(GSON_INSTANCE, "loot_modifiers");
+        instance = this;
+    }
+
+    public static LootModifierManager getInstance() {
```

Comment:
This system is completely messed up, you're creating a new instance every time a MinecraftServer is created, such as whenever a single player world is loaded.
Better to stop using a static context and instead give some other way of getting the manager.

Reply:
The only other option is to request it from the MinecraftServer instance. Which I didn't do as it would both make it annoying to get at or make the patch to MinecraftServer larger. I'm open to alternatives.

Reply:
What would it take to get the server instance to where it needs to be?
And the 'larger' patch would just be a 2 line accessor.

Reply:
MinecraftServer does not have an accessor except via a `World` object (and a `ServerWorld` at that). I'm fine going that route, but I'm in the dark about what's "best practice" here.

Reply:
If world accessible in context? Then ya, go through it. Going through several getters isn't that big of a deal.